web/server/entrypoint_test.go

```diff
@@ -192,3 +193,54 @@ func isExpectedShutdownError(err error, testLogger logging.Logger) bool {
 	testLogger.Errorw("Unexpected shutdown error", "err", err)
 	return false
 }
+
+// Tests that machine state properly reports initializing or running.
```

Comment:
can we add a test that tests the transition from initializing to running? I'm guess we may have to register a resource that takes a long time to start up but it would be good to have a test that fully tests the feature that you're adding.

but can discuss if it's difficult to do

Reply:
Another good question; it should be possible, yeah. I can at least delay the movement from initializing to running with a long-running constructor, but I think the test will be a bit racey.